src/ucp/core/ucp_context.h

```diff
@@ -334,6 +334,21 @@ ucp_tl_iface_latency(ucp_context_h context, const uct_iface_attr_t *iface_attr)
     return iface_attr->latency.overhead +
            (iface_attr->latency.growth * context->config.est_num_eps);
 }
+
+static UCS_F_ALWAYS_INLINE int
+ucp_tl_is_local_rsc(ucp_context_h context, ucp_rsc_index_t rsc_index)
```

Comment:
i hope we don't plan to use this function for BW calculation..

Reply:
actually planned - SHM transports are not affected by PPN, but network - does

Reply:
it should be encapsulated on UCT level

Reply:
why is it bad to rely on device type in UCP?

Reply:
it will be implemented as math formula where arguments will be defined depends from transport time